### PR TITLE
[scoped-custom-element-registry] Run `connectedCallback` on upgraded elements

### LINKED ISSUE #442: [scoped-custom-element-registry] lazy definitions seems to not upgrade the element if the same tag name has been defined in global registry
### Description

if we do a lazy component definition over a scoped registry using a tag name previously defined in the global registry, the components are not upgraded.

@kevinpschaaf 

### Example

See https://github.com/lazka/scoped-test-v2/blob/main/demo.js for a minimal example.

The `SomethingDifferent` component isn't visible if it is registered globally first and defineScopedElement() is called after the first update.

Two things:

Removing the first `customElements.define` fixes things
Calling `this.defineScopedElement` earlier (before render) fixes things as well

### Steps to reproduce

build the previous example and open the HTML inside the dist folder.

#### Expected behavior

It should upgrade the component

#### Actual behavior

The component is not upgraded

### Version

scoped-custom-element-registry v0.0.1

### PR DESCRIPTION
**Pull request by kevinpschaaf**
In the "upgrade" case (when an element is registered and there are elements in the `_awaitingUpgrade` map for that tag name), we were failing to run `connectedCallback()`. If the stand-in element definition was defined in the global registry first, then we would have no-opted the initial `connectedCallback()`, and so when manually upgrading the element later, we need to be sure to run its `connectedCallback` if `isConnected === true`.

Fixes #442

### CODE REVIEW

File: packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ if (!ShadowRoot.prototype.createElement) {
         this._awaitingUpgrade.delete(tagName);
         for (const element of awaiting) {
           pendingRegistryForElement.delete(element);
-          upgrade(element, definition);
```
Comment (justinfagnani) [inline packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js:91]: is there test coverage for this change, or were the tests already failing?

Comment (kevinpschaaf) [reply packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js:91]: Oh weird. I added a test... not sure what happened to it.

Comment (kevinpschaaf) [reply packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js:91]: Ok, added the test. PTAL.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -42,6 +42,27 @@ jobs:
       - name: WCT local (Chrome/Firefox)
         run: npm test
 
+  wtr-local:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
```
Comment (justinfagnani) [inline .github/workflows/tests.yml:50]: can we use `actions/setup-node@v2` and set `node-version: 14`?

File: web-test-runner.config.js [hunk 3]
```diff
@@ -6,26 +6,17 @@ const defaultBrowsers = [
   playwrightLauncher({product: 'firefox', concurrency: 1}),
 ];
 
-const envBrowsers = process.env.BROWSERS?.split(',').map((product) =>
-  playwrightLauncher({product})
-);
+const envBrowsers =
```
Comment (justinfagnani) [inline web-test-runner.config.js:9]: If we use Node 14 we don't need these changes

File: packages/scoped-custom-element-registry/test/scoped-registry.test.html.js [hunk 4]
```diff
@@ -54,5 +54,42 @@ describe('Scoped Registry', () => {
 
       expect(() => new CustomElementClass()).to.throw();
     });
+
+    it('should allow defining in global registry and scoped registery', () => {
+      const container = document.createElement('div');
+      document.body.appendChild(container);
```
Comment (justinfagnani) [inline packages/scoped-custom-element-registry/test/scoped-registry.test.html.js:60]: remove container at end of test

File: packages/scoped-custom-element-registry/test/scoped-registry.test.html.js [hunk 5]
```diff
@@ -54,5 +54,42 @@ describe('Scoped Registry', () => {
...
+      container.innerHTML = `<${tagName} attr="doc"></${tagName}`;
+      const docEl = container.firstChild;
+      expect(docEl).to.be.instanceOf(CustomElementClass);
+      expect(docEl.connected).to.be.true;
+      expect(docEl.attributeChanged).to.equal('attr:doc');
+
+      // Create scoped element, then upgrade via scoped registry
+      const registry = new CustomElementRegistry();
+      const shadowRoot = getShadowRoot(registry);
+      document.body.appendChild(shadowRoot.host);
```
Comment (justinfagnani) [inline packages/scoped-custom-element-registry/test/scoped-registry.test.html.js:86]: remove at end of test

**Review by justinfagnani — APPROVED** [review]

File: packages/scoped-custom-element-registry/test/scoped-registry.test.html.js (hunk 4 above)
Comment (kevinpschaaf) [reply packages/scoped-custom-element-registry/test/scoped-registry.test.html.js:60]: Done

File: packages/scoped-custom-element-registry/test/scoped-registry.test.html.js (hunk 5 above)
Comment (kevinpschaaf) [reply packages/scoped-custom-element-registry/test/scoped-registry.test.html.js:86]: Done

File: web-test-runner.config.js (hunk 3 above)
Comment (kevinpschaaf) [reply web-test-runner.config.js]: Will test whether it just works, didn't want to get sucked into a big task of debugging random infra stuff.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (kevinpschaaf) [reply .github/workflows/tests.yml]: Will test whether it just works, didn't want to get sucked into a big task of debugging random infra stuff.

**Pull request merged**